merkledag/node.go

```diff
@@ -134,6 +134,16 @@ func (n *Node) Copy() *Node {
 	return nnode
 }
 
+// UpdateNodeLink return a copy of the node with the link name set to point to
+// that. If a link of the same name existed, it is removed.
+func (n *Node) UpdateNodeLink(name string, that *Node) (*Node, error) {
+	newnode := n.Copy()
+	err := newnode.RemoveNodeLink(name)
+	err = nil // ignore error
+	err = newnode.AddNodeLink(name, that)
+	return newnode, err
```

Comment:
hmmm not sure if we want ipns to be mutable. I dont know this part of the codebase very well. I'll let @whyrusleeping shepherd the IPNS parts here